### ISSUE TITLE
Forward param 'is_true' will make the output different?

### ISSUE THREAD

**Issue by cjfan** [dmlc/mxnet#2867]
I use a net model with one BatchNorm layer and the outputs are not a loss function. 
when I call `executor.forward()` with 'is_train' set True or False, it give different output result,
I'm confused about it, no matter for training or for prediction, does the output should suppose to be same?
because for my trainning case, the output will also be compare with the label values to get the gradient, 
is there some different with the case for prediction?

And I find that, given same input, then continuously call the forward with 'is_train' set to True of False in arbitrary order, the output will only have two kind of result, exactly corresponding to the 'is_train' flag.
How to explain it?

**Comment by cjfan** [comment]
ok, batchnorm layer use different compute methods on mean and variance,
when do prediction, it only use a statistic of them.
so, at the begining, the output maybe clearly different, but with the model training done,
the difference will be very small.
